hopp/simulation/technologies/wind/wind_plant.py

```diff
@@ -186,19 +186,42 @@ def __attrs_post_init__(self):
                 system_model = Windpower.default(self.config_name)
             else:
                 # initialize system using pysam input file
-                input_file_path = resource_file_converter(self.config.model_input_file)
-                input_dict = load_yaml(input_file_path)
-
-                system_model = Windpower.new()
+                if isinstance(self.config.model_input_file,str):
+                    input_dict = load_yaml(self.config.model_input_file)
+                else:
+                    input_dict = self.config.model_input_file
+                nTurbs = 0
+                try:
+                    nTurbs = len(input_dict['Farm']['wind_farm_xCoordinates'])
+                except KeyError:
+                    pass
```

Comment:
This try/except block might need more logic or clarity. If `input_dict['Farm']['wind_farm_xCoordinates']` doesn't exist, then `nTurbs` is 0. Is that your desired outcome? If so, great! I'd suggest a minor change below:

```suggestion
                try:
                    nTurbs = len(input_dict['Farm']['wind_farm_xCoordinates'])
                except KeyError:
                    nTurbs = 0
```